### PR TITLE
Let 'GLookupFunc' and 'g_lru_cache_get' raise a 'GError'

### PR DESCRIPTION
**Pull request by arteymix**
This is useful for situations where the retrieval process can produce an error.

### CODE REVIEW

File: g-lru-cache.c [hunk 1]
```diff
@@ -183,11 +183,12 @@ g_lru_cache_get_size (GLruCache *self)
 }
 
 gpointer
-g_lru_cache_get (GLruCache *self, gpointer key)
+g_lru_cache_get (GLruCache *self, gpointer key, GError **error)
 {
 	g_return_val_if_fail (G_IS_LRU_CACHE (self), NULL);
 	
 	gpointer value;
+	GError *retrieve_error;
```
Comment (chergert) [inline g-lru-cache.c]: zeroing these are cheap, just assign to NULL here.

File: g-lru-cache.c [hunk 2]
```diff
@@ -220,7 +221,16 @@ g_lru_cache_get (GLruCache *self, gpointer key)
 			g_debug ("Retrieving value from external resource");
 #endif
 
-			value = self->priv->retrieve_func (key, self->priv->user_data);
+			retrieve_error = NULL;
```
Comment (chergert) [inline g-lru-cache.c]: drop this with above change

**Comment by arteymix** [comment]
It's fixed now! I tried to be somewhat fidel to https://developer.gnome.org/glib/stable/glib-Error-Reporting.html#GError

**Comment by chergert** [comment]
LGTM

**Pull request merged**